tests/unittests/test_events_primary_keys.py

```diff
@@ -5,28 +5,20 @@ class TestEventsPrimaryKeys(unittest.TestCase):
 
```

Comment:
We need a new unit test here to the `_generate_sdc_record_hash` to confirm that the hashing mechanism doesn't change, because if it does, that would indicate a major version bump and invalidation of all users' existing data.

It needs to include an example record with all data types as values, and confirmation of recursive sorting. Beyond the simple data types (string, int, float, etc), I'm thinking it should contain:
1. A list of simple values (out of sorted order)
2. A dictionary with keys out of order and simple values
3. A dictionary with lists as values with out of sorted order
4. A list with dictionaries as values that have keys sorted out of order

The last one is tricky, but the important part is confirming that lists that contain objects are also sorted in a guaranteed consistent representation.

I may have missed a case above, so please include whatever else you can come up with. If we miss a case here, it will cause duplicate data downstream and make the data useless.